### PR TITLE
IOSXR ntp global Module

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by ashwini-mhatre** [comment]
Module | statements | missing | excluded | branches | partial | coverage
-- | -- | -- | -- | -- | -- | --
Total | 249 | 10 | 0 | 97 | 9 | 95%
plugins/module_utils/network/iosxr/argspec/ntp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/ntp_global/ntp_global.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/config/ntp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/ntp_global/ntp_global.py | 71 | 0 | 0 | 51 | 1 | 99%
plugins/module_utils/network/iosxr/facts/ntp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/ntp_global/ntp_global.py | 37 | 1 | 0 | 14 | 0 | 98%
plugins/module_utils/network/iosxr/rm_templates/ntp_global.py | 48 | 7 | 0 | 30 | 7 | 82%
plugins/modules/iosxr_ntp_global.py | 14 | 1 | 0 | 2 | 1 | 88%
tests/unit/modules/network/iosxr/test_iosxr_ntp_global.py | 73 | 0 | 0 | 0 | 0 | 100%

File: plugins/module_utils/network/iosxr/facts/ntp_global/ntp_global.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        flatten_context_list = ["interface", "ntp"]
+
+        for x in flatten_context_list:
+            data = flatten_config(data, x)
+        # parse native config using the Ntp_global template
+        ntp_global_parser = Ntp_globalTemplate(
+            lines=data.splitlines(), module=self._module
+        )
+        objs = ntp_global_parser.parse()
+        if "access_group" in objs:
```
Comment (NilashishC) [inline plugins/module_utils/network/iosxr/facts/ntp_global/ntp_global.py:67]: We should sort all list of dictionaries for consistent rendering across different Python versions.

File: plugins/modules/iosxr_ntp_global.py [hunk 2]
```diff
@@ -0,0 +1,1077 @@
...
+              type: str
+            precedence:
+              description: Set precedence Please refer vendor document for valid entries.
+              type: str
+              choices: [ "critical", "flash", "flash-override", "immediate", "internet", "network", "priority", "routine" ]
+        ipv6: *ip
+        log_internal_sync:
+          type: bool
+          description: Logs internal synchronization changes.
+        master:
```
Comment (NilashishC) [inline plugins/modules/iosxr_ntp_global.py:152]: For consistency with IOS and NX-OS:
```yaml
master:
  stratum: 3
```

File: plugins/modules/iosxr_ntp_global.py [hunk 3]
```diff
@@ -0,0 +1,1077 @@
...
+          elements: dict
+          description: list of Key numbers for trusted time sources.
+          suboptions:
+            key_id:
+              type: int
+              description: Key numbers for trusted time sources.
+        update_calendar:
+          type: bool
+          description: Periodically update calendar with NTP time.
+        source:
```
Comment (NilashishC) [inline plugins/modules/iosxr_ntp_global.py:172]: If this is the source interface, can we name it as `source_interface`?

File: plugins/modules/iosxr_ntp_global.py (hunk 2 above)
Comment (KB-perByte) [reply plugins/modules/iosxr_ntp_global.py:152]: @ashwini-mhatre are the valid commands
`ntp master `and `ntp master 3` ?

File: plugins/module_utils/network/iosxr/facts/ntp_global/ntp_global.py (hunk 1 above)
Comment (ashwini-mhatre) [reply plugins/module_utils/network/iosxr/facts/ntp_global/ntp_global.py:67]: ack

File: plugins/modules/iosxr_ntp_global.py (hunk 2 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_ntp_global.py:152]: ack

File: plugins/modules/iosxr_ntp_global.py (hunk 3 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_ntp_global.py]: ack

**Review by KB-perByte — APPROVED** [review]

**Pull request merged**